app/src/userpages/modules/userPageStreams/actions.js

```diff
@@ -543,3 +559,31 @@ export const initNewStream = () => (dispatch: Function) => {
         uiChannel: false,
     }))
 }
+
+export const streamFieldsAutodetect = (id: StreamId) => (dispatch: Function) => {
+    dispatch(getStreamFieldAutodetectRequest())
+    return services.autodetectStreamfields(id)
+        .then((data) => ({
+            id,
+            config: {
+                fields: data.config.fields,
+            },
+        }))
+        .then(({ config: { fields } }) => {
+            if (!fields) {
+                throw new Error('No fields!')
+            }
+            dispatch(getStreamFieldAutodetectSuccess(fields))
+            Notification.push({
+                title: 'Fields autodetected!',
+                icon: NotificationIcon.CHECKMARK,
+            })
+        })
+        .catch((e) => {
```

Comment:
Use second argument to `.then(omSucces, onError)` instead of `.catch(onError)`, otherwise it will also catch errors in the success handler's `dispatch`